src/lib/mongoBackend/MongoCommonSubscription.cpp

```diff
@@ -202,33 +202,43 @@ void setEntities(const Subscription& sub, BSONObjBuilder* b)
   BSONArrayBuilder entities;
   for (unsigned int ix = 0; ix < sub.subject.entities.size(); ++ix)
   {
-    EntID en = sub.subject.entities[ix];
+    EntID       en        = sub.subject.entities[ix];
+    std::string finalId;
+    std::string finalType;
+    std::string isIdPattern;
+    bool        isTypePattern = false;
```

Comment:
I took a look at the entire file (without diff and stuff) and understood.

These local vars are used later, like this:
entities.append(BSON(CSUB_ENTITY_ID << finalId << CSUB_ENTITY_ISPATTERN << isIdPattern));

So no, definitely no easy way to change the type of isIdPattern